File: code/stable/gooltest/cpp/NameGenTest/NameGenTest.cpp
#include "NameGenTest.hpp"

#include <cassert>
#include <string>
#include <vector>

using std::string;
using std::vector;

int main(int argc, const char *argv[]) {
    vector<int> temp{1, 2, 3};
    vector<int> result(2);
    
    vector<int> temp0(0);
    for (int i = 1; i < 3; i++) {
        temp0.push_back(temp.at(i));
    }
    result = temp0;
    
    assert((int)(result.size()) == 2 && "Result list should have 2 elements after slicing.");
    
    assert(result.at(0) == 2 && "First element of result should be 2.");
    
    return 0;
}

void helper(vector<int> temp) {
    vector<int> result(2);
    
    vector<int> temp0(0);
    for (int i = 1; i < 3; i++) {
        temp0.push_back(temp.at(i));
    }
    result = temp0;
    
    assert((int)(result.size()) == 2 && "Result list should have 2 elements after slicing.");
}

File: code/stable/gooltest/csharp/NameGenTest/NameGenTest.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;

public class NameGenTest {
    
    public static void Main(string[] args) {
        List<int> temp = new List<int> {1, 2, 3};
        List<int> result = new List<int>(2);
        
        List<int> temp0 = new List<int>(0);
        for (int i = 1; i < 3; i++) {
            temp0.Add(temp[i]);
        }
        result = temp0;
        
        Debug.Assert( result.Count == 2 , "Result list should have 2 elements after slicing.");
        
        Debug.Assert( result[0] == 2 , "First element of result should be 2.");
    }
    
    private static void helper(List<int> temp) {
        List<int> result = new List<int>(2);
        
        List<int> temp0 = new List<int>(0);
        for (int i = 1; i < 3; i++) {
            temp0.Add(temp[i]);
        }
        result = temp0;
        
        Debug.Assert( result.Count == 2 , "Result list should have 2 elements after slicing.");
    }
}
